source/runtime_gui.cpp

```diff
@@ -711,12 +711,20 @@ void reshade::runtime::draw_gui()
 {
 	assert(_is_initialized);
 
+	bool show = _show_overlay;
+	api::input_source activation_source = api::input_source::none;
+
 	if (_input != nullptr)
 	{
 		if (_show_overlay && !_ignore_shortcuts && !_imgui_context->IO.NavVisible && _input->is_key_pressed(0x1B /* VK_ESCAPE */))
-			_show_overlay = false; // Close when pressing the escape button and not currently navigating with the keyboard
+			show = false; // Close when pressing the escape button and not currently navigating with the keyboard
 		else if (!_ignore_shortcuts && _input->is_key_pressed(_overlay_key_data, _force_shortcut_modifiers) && _imgui_context->ActiveId == 0)
-			_show_overlay = !_show_overlay;
+			show = !_show_overlay;
+
+		if (show != _show_overlay)
+		{
+			activation_source = api::input_source::keyboard;
+		}
```

Comment:
I think you can avoid this branch by just defaulting `activation_source` to `keyboard` instead of `none` above, since `activate_overlay`/`open_overlay` below won't be called if the state doesn't change anyway.